File: Backend/Domain/Models/State.cs
namespace Backend.Domain.Models
{
    public record class State
    {
        public Player Player { get; set; } = new();
        public ImportantItems ImportantItems { get; set; } = new();
        public Party Party { get; set; } = new();
        public DigimonBattle DigimonBattle { get; set; } = new();
        public CardBattle CardBattle { get; set; } = new();
        public Journal Journal { get; set; } = new();
    }
}

File: Frontend/src/models/state.ts
import type { ImportantItems } from "./important-items";
import type { Party } from "./party/party";
import type { DigimonBattle } from "./digimon-battle";
import type { CardBattle } from "./card-battle";
import type { Auctions } from "./auctions";
import type { Npcs } from "./npcs";
import type { Journal } from "./journal/journal";
import type { Player } from "./player";

export interface State {
  player: Player;
  importantItems: ImportantItems;
  party: Party;
  digimonBattle: DigimonBattle;
  cardBattle: CardBattle;
  auctions: Auctions;
  npcs: Npcs;
  journal: Journal;
}
